File: sdk/elixir/lib/rocksky/like.ex
defmodule Rocksky.Like do
  @moduledoc "`app.rocksky.like.*` endpoints. All require an authenticated client."

  alias Rocksky.HTTP

  @doc "Like a song. Body: `:uri`."
  def like_song(client, body),
    do: HTTP.procedure(client, "app.rocksky.like.likeSong", [], Map.new(body))

  @doc "Remove a like on a song. Body: `:uri`."
  def dislike_song(client, body),
    do: HTTP.procedure(client, "app.rocksky.like.dislikeSong", [], Map.new(body))

  @doc "Like a shout. Body: `:uri`."
  def like_shout(client, body),
    do: HTTP.procedure(client, "app.rocksky.like.likeShout", [], Map.new(body))

  @doc "Remove a like on a shout. Body: `:uri`."
  def dislike_shout(client, body),
    do: HTTP.procedure(client, "app.rocksky.like.dislikeShout", [], Map.new(body))
end

File: sdk/clojure/src/rocksky/like.clj
(ns rocksky.like
  "Endpoints under app.rocksky.like.*"
  (:require [rocksky.client :as c]))

(defn like-song
  "Like a song. Required: `:uri`."
  [client {:keys [uri] :as body}]
  (c/procedure client :app.rocksky.like.likeSong body))

(defn dislike-song
  "Dislike a song. Required: `:uri`."
  [client {:keys [uri] :as body}]
  (c/procedure client :app.rocksky.like.dislikeSong body))

(defn like-shout
  "Like a shout. Required: `:uri`."
  [client {:keys [uri] :as body}]
  (c/procedure client :app.rocksky.like.likeShout body))

(defn dislike-shout
  "Dislike a shout. Required: `:uri`."
  [client {:keys [uri] :as body}]
  (c/procedure client :app.rocksky.like.dislikeShout body))
